### DISCUSSION TITLE
How can I use _sum, using aggregations in a relation?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#16981] [Q&A]
```
model Numero_de_factura {
  id                 String               @id @default(uuid())
  numero             Int                  @unique @default(autoincrement())
  factura            Factura[]
}

model Factura {
  id                   String            @id @default(uuid())
  cantidad             Int
  total                Int
  numero_de_factura_id String
  numero_de_factura    Numero_de_factura @relation(fields: [numero_de_factura_id], references: [id])
}
```
Using the above code, I am trying to do something similar to the following (but I know include cannot be used with aggregates)
```
const total = await prisma.numero_de_factura.aggregate({
    where: {
      ...
      }
    },
    _sum: {
      total: true
    },
    include: {
      factura: true
    }
  })
```
What could be a workaround for this?

**Comment by nurul3101** [comment]
Hi @gmcamposano 👋 

Either you could use [Raw Query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access), or you can separate this query into two queries, one returning the sum and the other one returning the factura relations.

**Reply by gmcamposano** [reply]
1) I tried going to raw query route, and its a pain having to use "public" in eveything :)
2) Can you explain a little bit more how you would do two **separate** queries? because they are nested values. Each numero_de_factura has one or more facturas.

**Comment by sidd82** [comment]
You can do something like this
```
async function main() {
const orders = await prisma.order.findMany({
include: {
items: true
}
})

const totalAmount = orders.reduce((sum, order) => {
return sum + order.items.reduce((itemSum, item) => {
return itemSum + (item.quantity * item.price)
}, 0)
}, 0)

console.log("Total amount:", totalAmount)
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
